[youngseopark05] iP

youngseopark05 opened the pull request:
# HypeBot 🚀🎉

Welcome to **HypeBot**, your *Java-powered*, super-excited chatbot assistant! HypeBot helps you **crush your tasks**, keep up with your deadlines, and schedule events like a pro! Let’s get those tasks DONE and have a blast doing it! 💪🔥

## HypeBot Features 🌟

- **Track Tasks Like a Boss** 💼  
  You can track:
    - [x]   **ToDos**: Your everyday tasks, easy peasy!
    - [x]   **Deadlines**: Tasks with a due date—let’s beat the clock! ⏰
    - [x]   **Events**: Schedule events with start and end times. Get hyped! 🎉

- **Mark/Unmark Tasks Complete** ✔️❌  
  Celebrate those victories or update your progress as needed!

- **Find Tasks by Keyword** 🔍  
  Lost in the task list? No problem—HypeBot will help you find tasks with specific keywords!

- **Find Tasks by Date** 📅  
  Got a packed day? HypeBot can find all tasks and events happening on a specific date.

- **Delete Tasks** 🗑️  
  If you don't want to look at a task anymore, you shouldn't have to! 

## Getting Started 🏁

Here’s how you can start rocking your to-do list with HypeBot!

1. **Install HypeBot**  
   Download the executable JAR file from [here](https://github.com/youngseopark05/ip/releases/tag/v0.2) and run the command below in the folder you downloaded the JAR file to get HypeBot **_HOT AND RUNNING!_**

   ```terminal
   // Enter this on the terminal, you should be in the directory of where HypeBot was downloaded!
   java -jar hypebot.jar

2. **Ask for help**  
   There's nothing wrong with asking for help! 
   - To find out everything that HypeBot can do for YOU, type `help`


3. **Add a task**  
   Just tell HypeBot what you need to do, and it’ll handle the rest!

    - To add a **ToDo** type task, type in: `todo {task_name}`

    - To add a **Deadline** type task, type in: `deadline {task_name} /{due_date in yyyy-mm-dd}`

    - To add an **Event** type task, type in:
      `event {task_name} /{start_time in yyyy-mm-dd 24hh:mm} /{end_time in yyyy-mm-dd 24hhmm}`


4. **List all your tasks as one tasklist**  
   Wanna see all your tasks in a single view? HypeBot's got you covered!

    - To view all your added tasks as one tasklist, type in: `list`

## HypeBot's HYPEST Quirks 🙌

Here's some of HypeBot's functionalities to help you stay on TOP of things!

1. **Mark / Unmark a task as complete**  
   Ready to celebrate a win? Mark your tasks as complete like a champion!

    - To mark a task as complete, type in: `mark {task_number_in_list}
`
    - To unmark a task as complete, type in: `unmark {task_number_in_list}`


2. **Search for tasks by keyword**  
   Can’t find a task? No worries, HypeBot’s got you!

    - To search for tasks containing keywords, type in: `find {keyword1} {keyword2} ...`

   HypeBot will search for tasks matching ANY of the keywords, so you can search for multiple if you want to!


3. **Find tasks by date**  
   What’s happening on a specific day? Let HypeBot tell you!
   - To search for tasks happening on a certain day, type in: `happening {search_date in yyyy-mm-dd}`


4. **Use convenient dates**  
   No time to put in all those pesky dates? No worries! HypeBot will figure out the dates for you if you type in
   `today`, `tomorrow`, and any days of the week, where a date entry should be! (You'll still have to put in a time
   yourself though!)


5. **Delete tasks**  
   Say "BYE BYE BYE!" faster than you can say NSYNC when you don't need a task anymore!
   - To delete a task, type in: `delete {task_number_in_list}`
   - To delete ALL your tasks, type in: `deleteall`

## So you think you know how to use HypeBot? ⏰

Once you become a familiar user, you might want to start getting your tasks done at even BLAZING speeds - HypeBot's got 
YOUR back!   
Use any of the following shortcuts for an even FASTER experience - be done with your tasks like clockwork!
- `help` --> `h`
- `todo` --> `td`
- `deadline` --> `dl`
- `event` --> `ev`
- `list` --> `l`
- `mark` --> `m`
- `unmark` --> `u`
- `delete` --> `d`
- `deleteall` -> `da`
- `find` --> `f`
- `happening` --> `hp`
- `bye` --> `b`

Here's some more date-related shortcuts too:

- `today` --> `tdy`
- `tomorrow` --> `tmw`, `tmrw`
- `monday` --> `mon`
- `tuesday` --> `tue`, `tues`
- `wednesday` --> `wed`
- `thursday` --> `thu`, `thur`, `thurs`
- `friday` --> `fri`
- `saturday` --> `sat`
- `sunday` --> `sun`

## What are you waiting for? 🤖

> The only way to do great work is to love what you do." — Let’s get that task list in shape! 💪

With HypeBot, every task feels like a victory lap! 🏁 Let’s CRUSH those goals together! 💥

LET'S GO!!! 🎉🚀

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

youngseopark05 committed:
Rename main class to 'HypeBot'

Main class name is the default 'Duke'.

The lack of a distinct identity between this new chatbot
and the original Duke makes it hard to differentiate
between the two.

Giving this chatbot a distinct identity allows us to
have an emotional attachment and dedication to
producing a higher quality product. It also allows
users to form an emotional attachment to the
chatbot so they can enjoy spending time interacting
with it.

Let's rename this chatbot 'HypeBot'.

Renaming this chatbot to reflect its desired personality
of a cheerful, motivating companion sets users'
expectations on how their interactions with the chatbot
are going to feel.

youngseopark05 committed:
README: Update name in title from 'Duke' to 'HypeBot'

youngseopark05 committed:
Add greeting and exit methods

There is only a greeting that is triggered in the
main() method.

The lack of an exit and the lack of separation
between a greeting and an exit complicate the
start and end of the conversation between the
user and the bot.

Extracting the greeting and exit behaviour
each allows us to control the interface at the
start and end of the conversation between the
user and the bot.

Let's make a greet() method that welcomes the
user and and exit() method that wishes them
goodbye.

Separating the greeting and exit message ensures
they can be called separately.

youngseopark05 committed:
Reformat logo and greeting

Logo is multiple concatenated strings,
while greeting does not directly prompt
user for a command.

Not being able to visualise the logo
as a single string block makes adjusting
mistakes in ASCII art difficult, while
not directly prompting user for a command
could confuse user and prevent them
from initiating interaction with HypeBot.

Let's convert the logo into a single
string block using \s and prompt the user
for a command.

Refer to Ghost font provided by
https://www.asciiart.eu/text-to-ascii-art

youngseopark05 committed:
Add support for echo

HypeBot has no support for accepting user
input or reacting to it.

HypeBot needs to be able to accept user
input and have some kind of input handling
so that when the user types in some form of
exit command, HypeBot is able to act accordingly
by calling the exit() method.

Let's use a Scanner and a while loop that
constantly checks whether the exit
command 'bye' in any form of
capitalisation to exit, otherwise
repeat the user's input allows us to
achieve basic echo functionality.

Using a temporary boolean variable as
the condition for the input loop
allows us to store the command entered,
process it into lowercase so we don't
have to test for all combinations which
the command 'bye' can be entered.

youngseopark05 committed:
Add support for adding and listing tasks

youngseopark05 committed:
Add Task class with name and completion status

youngseopark05 committed:
Extracted add(), list() methods and added mark(), unmark() methods

youngseopark05 committed:
Add ToDo.java which inherits Task.java

youngseopark05 committed:
Add Deadline.java which inherits Task.java, has a new attribute dueDate

youngseopark05 committed:
Add Event.java which inherits from Task.java, has additional startTime and endTime

youngseopark05 committed:
Change add message to show number of tasks user has

youngseopark05 committed:
Reformat Task.toString() so there is no newline at the end

youngseopark05 committed:
Add support for different types of Tasks (ToDos, Deadlines, Events)

Refactor adding buffer lines to their own
methods

Reformat input handling into switch-case
statement with more comprehensive error
handling

youngseopark05 committed:
Reformat input handling and display of dates and times using StringBuilder

youngseopark05 committed:
Add error handling for empty ToDo task name

youngseopark05 committed:
Rename 'Duke' in runtest.bat to 'HypeBot'

youngseopark05 committed:
Add tests for expected behavior for all existing commands

youngseopark05 committed:
Add support for deletion of tasks

youngseopark05 committed:
Add tests for delete command

youngseopark05 committed:
Add more tests for delete command

youngseopark05 committed:
Add Javadoc and reformat to SE-EDU coding style

waihongteh commented on src/main/java/HypeBot.java:
```diff
@@ -0,0 +1,230 @@
+            String taskName = taskNameBuilder.toString();
+
+            switch (command) {
+                case "bye":
```
Oops, case should not have indentation according to CS2103T coding standard. Similar goes across the switch statements.

waihongteh commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+    public Deadline(String name, String dueDate) {
+        super(name);
+        String[] dateParts = dueDate.split(" ");
+        StringBuilder dateSB = new StringBuilder(dateParts[0]).append(": ");
```
Perhaps acronyms should be lowercase like dateSb when used as a part of name?

waihongteh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,28 @@
+        super(name);
+
+        String[] startParts = startTime.split(" ");
+        StringBuilder startSB = new StringBuilder(startParts[0]).append(": ");
```
startSb instead of startSB since it is an abbreviation?

youngseopark05 replied:
Thanks for pointing that out.

waihongteh commented on src/main/java/HypeBot.java:
```diff
@@ -0,0 +1,230 @@
+
+public class HypeBot {
+    private static final String BUFFER_LINE = "______________________________________________________________________\n";
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
```
Maybe tasks should be in lowercase since it is constantly changing? Like the content is changing even though the pointer remains the same? So might not be classified fully as a CONSTANT.

youngseopark05 replied:
Thanks for in the input, I agree.

waihongteh commented on src/main/java/HypeBot.java:
I like how clean and structured the code is!

youngseopark05 replied:
Thank you!

waihongteh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,28 @@
+        this.startTime = startSB.toString();
+
+        String[] endParts = endTime.split(" ");
+        StringBuilder endSB = new StringBuilder(endParts[0]).append(": ");
```
endSb instead of endSB since it is an abbreviation?

waihongteh reviewed (commented):
Overall the code is very clean and structured. Nice!

youngseopark05 committed:
Add save and load functionality to tasks entered by user

youngseopark05 committed:
Move Task.java and its children to package 'task'

youngseopark05 committed:
Add JavaDoc to all classes in package 'task' and some comments to HypeBot.java

youngseopark05 committed:
Merge branch 'branch-Level-7'

youngseopark05 committed:
Change Deadline dueDate, Event startTime and endTime to LocalDateTime

youngseopark05 committed:
Add error handling for scenarios where user does not input dates or times in desired format

youngseopark05 committed:
Change Deadline dueDate to LocalDate

youngseopark05 committed:
Add functionality for 'happening' command for user to access events happening on given date

youngseopark05 committed:
Merge branch 'branch-Level-8'

youngseopark05 committed:
Add classes to reinforce OOP Design

youngseopark05 committed:
Fix error message for mark/unmark/delete commands when index out of bounds

youngseopark05 committed:
Add 'help' command functionality to list out all possible commands for user

youngseopark05 committed:
Add tests for 'happening' and 'help' command

youngseopark05 committed:
Merge branch 'add-gradle-support'

youngseopark05 committed:
Integrate Gradle into the project

youngseopark05 committed:
Add date validation for Deadline and Event tasks to prevent illogical start times, end times, and due dates

youngseopark05 committed:
Add JUnit5 tests for ToDo, Deadline, and Event methods

youngseopark05 committed:
Fix bug leaving startTime and endTime null

youngseopark05 committed:
Add data/tasks.txt for HypeBot to access when running on other clients

youngseopark05 committed:
Fix bug where HypeBot continues to respond even with no input

youngseopark05 committed:
Create executable JAR file

youngseopark05 committed:
Fix error in creating tasklist when running app with a JAR

youngseopark05 committed:
Add JavaDoc to HypeBot methods

youngseopark05 committed:
Add JavaDoc to Tasklist methods

youngseopark05 committed:
Add JavaDoc to Ui methods

youngseopark05 committed:
Add JavaDoc to public methods in hypebot.task package

youngseopark05 committed:
Add JavaDoc to hypebot.storage methods

youngseopark05 committed:
Add JavaDoc to hypebot.command methods

youngseopark05 committed:
Add JavaDoc to Parser methods and Messages class

youngseopark05 committed:
Clean up code to adhere to SE-EDU Java Standard

youngseopark05 committed:
Add 'find' command functionality that searches for tasks containing keywords specified by user

youngseopark05 committed:
Merge branch 'branch-A-JavaDoc'

youngseopark05 committed:
Merge branch 'branch-A-CodingStandard'

youngseopark05 committed:
Fix bug in parsing keywords that resulted in no matching results

youngseopark05 committed:
Merge branch 'branch-Level-9'

youngseopark05 committed:
Remove unused import statement from Tasklist.java

youngseopark05 committed:
Delete unnecessary comment from Tasklist.java

youngseopark05 committed:
Fix coding style violations using CheckStyle

youngseopark05 committed:
Merge branch 'A-CheckStyle'

youngseopark05 committed:
Fix coding style violations using CheckStyle

youngseopark05 committed:
Add autosave and autodelete outdated functionality

youngseopark05 committed:
Add deleteall functionality, separate Parser into different classes,
check for duplicates when adding Tasks, and add support for aliases
when entering dates (e.g. today, tomorrow, etc.).

The Parser currently handles all user input and error handling for user
input. This is violation of SLAP.

Separating Parsers for different types of parsing helps reinforce OOP
principles and keep better track of error handling across different
across different errors encountered.

Let's extract a CommandParser to make sense of which Commands to be
executed by the system, a DateTimeParser to parse dates and times,
an IndexParser to make sense of indexes entered, and a TaskParser to
make sense of information entered into Task objects.

Following SLAP is important for code readability, quality, and ease for
debugging when encountering unexpected errors.

youngseopark05 committed:
Add GUI functionality

youngseopark05 committed:
Merge branch 'branch-Level-10'

youngseopark05 committed:
Add assertions to check tasklist file exists or aliases parsed correctly

youngseopark05 committed:
Merge pull request #2 from youngseopark05/branch-A-Assertions

Add assertions to check tasklist file exists or aliases parsed correctly

youngseopark05 committed:
Merge branch 'master' of https://github.com/youngseopark05/ip

youngseopark05 committed:
Merge branch 'master' into branch-A-CodeQuality

youngseopark05 committed:
Merge branch 'master' into branch-A-Streams

youngseopark05 committed:
Make Parsers more OOP, handle errors more fully to be more informative

youngseopark05 committed:
Refactor Parsers into more descriptive packages

youngseopark05 committed:
Update EventTest and DeadlineTest methods

youngseopark05 committed:
Use Stream::anyMatch and Stream::filter and handle more errors for
'find' and 'happening' command

youngseopark05 committed:
Add JavaDoc and code quality improvements to adhere to CheckStyle

youngseopark05 committed:
Merge pull request #3 from youngseopark05/branch-A-CodeQuality

Add well-annotated JavaDoc and code quality improvements

youngseopark05 committed:
Merge branch 'master' of https://github.com/youngseopark05/ip

youngseopark05 committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/hypebot/command/FindCommand.java
#	src/main/java/hypebot/command/HappeningCommand.java
#	src/main/java/hypebot/tasklist/Tasklist.java

youngseopark05 committed:
Fix bug with ToDo::toFileString() encoding wrong format

ToDo::toFileString() encodes the UI Task representation to a save file
instead of the intended file format by calling super.toString() instead
of toFileString().

This causes an error in decoding tasks at the booting of HypeBot by the
TasklistDecoder as Tasks that should be saved and decodeable become
encoded in a wrong format, not decodeable, then not saveable.

Change super.toString() to super.toFileString().

This will allow proper encoding and decoding.

youngseopark05 committed:
Fix bug with Tasklist::getHappeningOn() and Tasklist::getNameContains()

Tasklist::getHappeningOn and Tasklist::getNameContains are converting
Object objects returned by the ArrayList<>.clone() method to Tasklist
objects.

This causes an error as Tasklist, as hard type conversion of Objects to
Tasklists, a mutable List, are not permitted.

Restructure code so that Object returned by ArrayList<>.clone() is
converted into a List<Task> with unchecked warnings suppressed, then use
ArrayList<>.addAll() to add elements to a new Tasklist containing
filtered results.

This will stop the error from occurring.

youngseopark05 committed:
Fix bugs in UiTaskParser

youngseopark05 committed:
Merge pull request #4 from youngseopark05/branch-A-Streams

Incorporate stream functions for 'find' and 'happening' commands

youngseopark05 committed:
Merge branch 'master' of https://github.com/youngseopark05/ip

youngseopark05 committed:
Create main.yml

youngseopark05 committed:
Fix CheckStyle errors

youngseopark05 committed:
Clarify shortcuts available in help message

youngseopark05 committed:
Merge branch 'branch-BCD-Extension'

youngseopark05 committed:
Update profile pictures for user and HypeBot

youngseopark05 committed:
Change color scheme of MainWindow and DialogBox

youngseopark05 committed:
Add more custom messages

youngseopark05 committed:
Update README.md so that it includes all HypeBot features

youngseopark05 committed:
Update README.md to match user guide

youngseopark05 committed:
A-Release

youngseopark05 committed:
Fix JAR file download link in user guide

youngseopark05 committed:
Merge branch 'master' of https://github.com/youngseopark05/ip

youngseopark05 committed:
Revert "A-Release"

This reverts commit dd2440c29fc3c0e6a0eae13a03bc4c3466348a3d.

youngseopark05 committed:
Fix link in user guide for JAR

youngseopark05 committed:
Fix line break in README.md

youngseopark05 committed:
Update README.md

youngseopark05 committed:
Update README.md

youngseopark05 committed:
Merge branch 'master' of https://github.com/youngseopark05/ip
